src/javascript/crypto/e2e/coname/lookup_verify.js

```diff
@@ -562,41 +360,60 @@ e2e.coname.verifyLookup = function(cfg, user, pf, now) {
         'VerifyLookup: VRF verification failed');
   }
 
-  root = e2e.coname.verifyConsensus_(realm, pf.ratifications, now);
+  if (!e2e.coname.verifyConsensus_(realm, pf.ratifications, new Date())) {
+    throw new e2e.error.InvalidArgumentsError(
+        'VerifyConsensus: insufficient signatures');
+  }
+
+  // reconstruct the partial merkle tree
+  tree = e2e.coname.reconstructTree_(
+      pf.tree_proof,
+      e2e.coname.toBits_(e2e.coname.MERKLE_INDEX_BITS, pf.index));
 
-  verifiedEntryHash = e2e.coname.reconstructTreeAndLookup_(
-      realm.tree_nonce || [], root, pf.index, pf.tree_proof);
+  // recompute the hash of the constructed tree
+  rootHash = e2e.coname.recomputeHash_(realm.tree_nonce || [], [], tree);
 
-  if (!verifiedEntryHash) {
+  // since the hash is already verified, the merkle lookup is unneccessary
+  // compare root hash with the recomputed hash of the constructed tree
+  if (!e2e.compareByteArray(
+      rootHash,
+      pf.ratifications[0].head.head.root_hash)) {
+    throw new e2e.error.InvalidArgumentsError(
+        'VerifyLookup: failed to verify merkle tree: Root hashes mismatch!');
+  }
+
+  if (!pf.tree_proof.existing_entry_hash) {
     if (!pf.entry) {
       throw new e2e.error.InvalidArgumentsError(
-          'VerifyLookup: non-empty entry ' + pf.entry +
+          'VerifyLookup: non-empty entry ' + JSON.stringify(pf.entry) +
           ' did not match verified lookup result <null>');
     }
     if (!pf.profile) {
       throw new e2e.error.InvalidArgumentsError(
-          'VerifyLookup: non-empty profile ' + pf.profile +
+          'VerifyLookup: non-empty profile ' + JSON.stringify(pf.profile) +
           ' did not match verified lookup result <null>');
     }
-    return false;
-
-  } else {
+    return true;
```

Comment:
This is changed from `false` to `true`, meaning that the profile is verified to be empty.